pyiele/__main__.py

```diff
@@ -0,0 +1,169 @@
+        config.target_directory = args.build_dir
+        config.master_key = args.master_key
+        config.passphrase = args.passphrase
+
+        if(args.file is None):
+            if os.path.isdir(config.tests_dir):
+                for child_file in os.listdir(config.tests_dir):
+                    if child_file.endswith(".sol"):
+                        run_test_file(config.tests_dir + "/" + child_file)
+            elif os.path.isfile(config.test_dir):
+                    if child_file.endswith(".sol"):
+                        run_test_file(child_file)
+        else:
+            file_path = args.file
+            if (os.path.isdir(file_path)):
+                for child_file in os.listdir(file_path):
+                    if child_file.endswith(".sol"):
+                        run_test_file(file_path + "/" + child_file)
+            elif (os.path.isfile(file_path) and (file_path).endswith(".sol")):
+                run_test_file(file_path)
+            else:
+                fatal("Invalid file: "+ file_path)
+        print("==  Passing Tests:", config.passing_tests)
+        print("==  Failing Tests:", config.failing_tests)
+
+    elif args.command == 'blackbox':
+        config.port = args.port
+        config.coverage_timeout = timeout=args.timeout
+        config.master_key = args.master_key
+        config.passphrase = args.passphrase
+
+        notif("Collecting data for blackbox random testing...")
+
+        # Retrive coverage_report,contract_map
+        coverage_report = get_coverage_report()
+        contract_map = get_contract_map(config.target_directory)
+
+        # Extract contracts that are available for blackbox testing
+        available_contract = get_available_contract(coverage_report, contract_map)
+
+        notif("Preparation finished, altogether " + str(len(available_contract)) + " contacts to test")
+        notif("")
+
+        # Process contracts in alphabetical order
+        available_contract = dict(sorted(available_contract.items()))
+        all_calls = []
+        all_constructors = {}
+        for solidity_file in available_contract:
+            contract_json = config.target_directory + "/" + get_file_name(solidity_file) + ".json"
+            solidity_functions = fetch_function_data(contract_json)
+            for contract in solidity_functions:
+                if(not contract["contractName"] in list(available_contract.keys())):
+                    continue
+                contract_calls, constructor_info = blackbox_test_single_contract(contract, available_contract)
+                all_calls.extend(contract_calls)
+            if len(constructor_info) == 0:
+                all_constructors[solidity_file] = []
+            else:
+                all_constructors[solidity_file] = constructor_info[0]["input"]
+
+        # Write founded tests to blackbox-random.json
+        with open('blackbox-random.json', 'w') as outfile:
+            json.dump(all_calls, outfile)
```

Comment:
How is this file used? Is it incorporated into the web report too?

Reply:
AFAIK, this file is not used in the web report currently.